presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java

```diff
@@ -49,38 +49,67 @@ private KafkaQueryRunner() {}
     private static final Logger log = Logger.get(KafkaQueryRunner.class);
     private static final String TPCH_SCHEMA = "tpch";
 
-    static DistributedQueryRunner createKafkaQueryRunner(TestingKafka testingKafka, TpchTable<?>... tables)
-            throws Exception
+    public static Builder builder(TestingKafka testingKafka)
     {
-        return createKafkaQueryRunner(testingKafka, ImmutableList.copyOf(tables));
+        return new Builder(testingKafka);
     }
 
-    static DistributedQueryRunner createKafkaQueryRunner(TestingKafka testingKafka, Iterable<TpchTable<?>> tables)
-            throws Exception
+    public static class Builder
+            extends DistributedQueryRunner.Builder
     {
-        return createKafkaQueryRunner(testingKafka, tables, ImmutableMap.of());
-    }
+        private final TestingKafka testingKafka;
+        private Map<String, String> extraKafkaProperties = ImmutableMap.of();
+        private Iterable<TpchTable<?>> tables = ImmutableList.of();
+        private Map<SchemaTableName, KafkaTopicDescription> extraTopicDescription = ImmutableMap.of();
+
+        protected Builder(TestingKafka testingKafka)
+        {
+            super(testSessionBuilder()
+                    .setCatalog("kafka")
+                    .setSchema(TPCH_SCHEMA)
+                    .build());
+            this.testingKafka = requireNonNull(testingKafka, "testingKafka is null");
+        }
 
-    static DistributedQueryRunner createKafkaQueryRunner(TestingKafka testingKafka, Iterable<TpchTable<?>> tables, Map<SchemaTableName, KafkaTopicDescription> topicDescription)
-            throws Exception
-    {
-        return createKafkaQueryRunner(testingKafka, ImmutableMap.of(), tables, topicDescription);
+        public Builder setExtraKafkaProperties(Map<String, String> extraKafkaProperties)
+        {
+            this.extraKafkaProperties = extraKafkaProperties;
+            return this;
+        }
+
+        public Builder setTables(Iterable<TpchTable<?>> tables)
+        {
+            this.tables = tables;
+            return this;
+        }
+
+        public Builder setExtraTopicDescription(Map<SchemaTableName, KafkaTopicDescription> extraTopicDescription)
+        {
+            this.extraTopicDescription = extraTopicDescription;
+            return this;
+        }
+
+        @Override
+        public DistributedQueryRunner build()
+                throws Exception
+        {
+            Logging logging = Logging.initialize();
```

Comment:
move it back to `createKafkaQueryRunner`?

Reply:
No because entire `DistributedQueryRunner.Builder.build()` will not produce correct log messages.